### PR TITLE
Add cloudwatch e2e tests

### PR DESCRIPTION
**Pull request by LiliC**
First step towards having all the e2e tests in go! 🎉 

Closes https://github.com/weaveworks/launcher/issues/228

Note: Decided to just add a creation test, as the deletion one took a really long time around 5-10 minutes to delete all the resources, not sure if the benefits are there for that?

### CODE REVIEW

File: integration-tests/e2e/cloudwatch_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	// Create ConfigMap
+	cm := test.CreateConfigMapFromFile(ns, "cloudwatch-cm.yaml")
+
+	// Wait for the above resources to be created.
+	test.WaitForSecretReady(secret, timeout)
+	test.WaitForConfigMapReady(cm, timeout)
+
+	// Make sure that the Cloudwatch deployment with name 'cloudwatch-exporter' in 'weave' ns exists.
+	_, err := test.GetDeployment(ns, cloudwatchDeploymentName)
```
[the remark this answers is not in the document]
Comment (LiliC) [reply integration-tests/e2e/cloudwatch_test.go]: We already wait for the `ConfigMapReady` above, so when both of those objects are created the deployment should be as well. I understand your point of view, but works now? If it stops or gives us false results we can change it to `WaitForDeploymentByName`? Up to you.

File: Gopkg.toml [hunk 2]
```diff
@@ -88,3 +87,8 @@ branch = "release-1.9"
 [[constraint]]
   name = "github.com/prometheus/client_golang"
   version = "v0.9.0-pre1"
+
+[[constraint]]
```
[the remark this answers is not in the document]
Comment (LiliC) [reply Gopkg.toml]: Yes good point, was thinking of actually pining it to a specific commit, as the API might change in the future, WDYT?

File: .circleci/config.yml [hunk 3]
```diff
@@ -127,6 +127,61 @@ jobs:
...
 
+  integration_cloudwatch:
+    machine: true
+    environment:
+      CHANGE_MINIKUBE_NONE_USER: true
+    working_directory: ~/.go_workspace/src/github.com/weaveworks/launcher
+    steps:
+      - checkout
+      - run: make dep
+      - &integration_deps
```
[the remark this answers is not in the document]
Comment (LiliC) [reply .circleci/config.yml]: I just copy pasted from another test setup, but agree.

File: integration-tests/e2e/cloudwatch_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlespiau) [reply integration-tests/e2e/cloudwatch_test.go]: huum, we are waiting for the initial configmap to be ready (one of the launcher-agent inputs), not the configmap we are installing with the cloudwatch-exporter. We can definitely wait and see if the test fails at some point :)

File: Gopkg.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlespiau) [reply Gopkg.toml]: that's what the `.lock` file does IIUC, `dep ensure` will always take the same commit (the one in the .`lock` file) until someone does a `dep ensure -upgrade github.com/dlespiau/kube-harness`.

File: integration-tests/e2e/cloudwatch_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (LiliC) [reply integration-tests/e2e/cloudwatch_test.go]: Yes, exactly. These are waiting for the user created resources to be ready, which are picked up by the agent when we watch for them and apply the deployment and configmap, so when they are created the agent should create those resources. Makes sense for me, but if you have a better suggestion?

File: Gopkg.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (LiliC) [reply Gopkg.toml]: Yes, I know! :) But if someone deletes the `.lock` file....

[the remark this answers is not in the document]
Comment (dlespiau) [reply Gopkg.toml]: hum, hadn't thought about someone delete the `.lock` file! Huum, yes, pinning to a commit sounds good!

File: integration-tests/e2e/cloudwatch_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlespiau) [reply integration-tests/e2e/cloudwatch_test.go]: I still think there's a race: we `WaitForConfigMapReady(cm)` and the next line we `GetDeployment(ns, cloudwatchDeploymentName)` checks launch-generator has created the deployment. If it takes a bit long to download the manifests, the deployment may not be there. Feel free to merge as is though!

**Comment by LiliC** [comment]
@dlespiau Addressed your comments, PTALA, thanks!

**Comment by dlespiau** [comment]
Thanks! lgtm!

**Pull request merged**